utils/fix-rules.py

```diff
@@ -10,6 +10,9 @@
 import ssg
 
 
+SSG_ROOT = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
```

Comment:
I'm starting to think that `ssg_root` should a property retrievable from the build_config.

Reply:
Note that this is a default value and has a flag to change the value used at runtime. This tool can be used on an external SSG tree if desired.

`build_config.yaml` needs to be templated (adding in `SSG_ROOT`), requiring the build to be done _first_ prior to running these utilities.

However, the intent of these utilities (well, `fix_rules` and my other three I just opened PRs for) was to run without requiring a build to be done. They operate on rule YAMLs, not XCCDF/OVAL/... other build artifacts... so in my view, requiring the `build_config.yml` to already be templated is... eh? 

Thoughts?

Reply:
> Note that this is a default value and has a flag to change the value used at runtime. This tool can be used on an external SSG tree if desired.

Are you doing builds from an external tee at moment? This is something that we aim for in the long run, to have a better decoupling between build system and content tree.

> 
> `build_config.yaml` needs to be templated (adding in `SSG_ROOT`), requiring the build to be done _first_ prior to running these utilities.
> 
> However, the intent of these utilities (well, `fix_rules` and my other three I just opened PRs for) was to run without requiring a build to be done. They operate on rule YAMLs, not XCCDF/OVAL/... other build artifacts... so in my view, requiring the `build_config.yml` to already be templated is... eh?
> 
> Thoughts?

I get your point, but other "core" build time tools could benefit from a somehow centralized `ssg_root`.

Reply:
> Are you doing builds from an external tee at moment? This is something that we aim for in the long run, to have a better decoupling between build system and content tree.

Right now we're just starting a new, private fork of CaC and are working to identify which parts we want to upstream. Most of the tooling in the other PRs is to make maintaining the private fork easier, but is other of general use to the community, so upstreaming it makes sense. We're still discussing what all we're going to upstream; I'm hoping it is generally useful stuff though.

This PR was to make the diffs more manageable ;-)

We're likely going to start by building CaC internally and placing the resulting artifacts in our internal git repo where our existing content is (replacing our existing artifacts with CaC-built artifacts). So an in-tree build is fine for what we're doing now.

How far are you interested in going with decoupling the build system? I mentioned to some internal users that the CaC build system could probably build their CVE OVALs (think a multi-distribution collaborative CVE database for linux distros' packages -- is Red Hat interested?) but there's no commitment to that on our end. Just idle thoughts :-)

However....

I worry that our asset formats aren't yet "standardized" enough to make this possible. There's a lot of experimentation that I think is still happening (think control files) that would be much harder by this approach, especially if we had to coordinate with another project. I don't know if we will ever get to that point, but just pointing out that even within the current project... ...version A of content and version B of content wouldn't necessarily all be buildable by version A (or version B's) build system. So you might as well keep a tight coupling rather than a loose one for the time being. Outside of development scenarios that is. 

I think forking the build system is the more attractive option in that case, at least, until one or both projects mature and decide that the build systems should be re-joined or that they're completely separate ideas. I dunno.

Reply:
This is not something to be solved in this PR.